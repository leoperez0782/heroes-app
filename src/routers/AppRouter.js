import React, { useContext } from 'react';
import {
    BrowserRouter as Router,
    Switch,
} from "react-router-dom";
import { AuthContext } from '../auth/AuthContext';

import { LoginScreen } from '../login/LoginScreen';
import { DashBoardRouter } from './DashBoardRouter';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';


export const AppRouter = () => {

    const {user} = useContext(AuthContext);
    return (
        <Router>
           

            <div>
                <Switch>
                    <PublicRoute 
                    exact 
                    path="/login" 
                    component={LoginScreen}
                    isAuthenticated={user.logged} />
                    {/* <Route exact path="/login" component={LoginScreen} /> */}
                    
                    <PrivateRoute path="/" 
                    component={ DashBoardRouter } 
                    isAuthenticated = {user.logged}/>
                </Switch>
            </div>


        </Router>
    );
}
