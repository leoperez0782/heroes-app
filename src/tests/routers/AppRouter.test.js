import { mount } from "enzyme"
import { AuthContext } from "../../auth/AuthContext";
import { AppRouter } from '../../routers/AppRouter';

describe('Pruebas en <AppRouter />', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    };
    test('debe de mostrar el login si no esta autenticado', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>

        );

        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar el componente de marvel si esta autenticado', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Lolo'
            }
        };
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>

        );

        expect(wrapper.find('.navbar').exists()).toBe(true);

    })


})
