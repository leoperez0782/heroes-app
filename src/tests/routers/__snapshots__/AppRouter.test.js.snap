// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el login si no esta autenticado 1`] = `
<div>
  <div>
    <h1>
      Login Screen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
